### PR TITLE
cluster deployments: teardown on cancel

### PR DESCRIPTION
**Pull request by boz**
* provider/cluster: teardown on deployment/fulfillment close.
* provider/manifest: allow multiple leases per manifest.
* provider: handle multiple manifest groups.
* provider: match Manifest and Deployment groups via Name field.
* types: add `GroupSpec.Name`, `DeploymentGroup.Name` fields.
* sdl: populate `GroupSpec.Name` field.

### CODE REVIEW

File: provider/cluster/mocks/client.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
+// Code generated by mockery v1.0.0
```
Comment (golangci) [inline provider/cluster/mocks/client.go]: package comment should be of the form "Package mocks ..."

File: provider/cluster/monitor.go [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+	"fmt"
+
+	lifecycle "github.com/boz/go-lifecycle"
+	"github.com/ovrclk/akash/types"
+	"github.com/tendermint/tmlibs/log"
+)
+
+type deploymentState string
+
+const (
```
Comment (golangci) [inline provider/cluster/monitor.go]: only the first constant has an explicit type

File: provider/cluster/service.go [hunk 3]
```diff
@@ -109,22 +123,101 @@ loop:
...
+				if err := state.monitor.update(mgroup); err != nil {
+					s.log.Error("updating deployment", "err", err)
+				}
+
+			case *event.TxCloseDeployment:
+
+				// teardown/undeploy managed deployments
+
+				for _, dm := range s.deployments {
+					if bytes.Compare(ev.Deployment, dm.reservation.OrderID().DeploymentID()) == 0 {
```
Comment (golangci) [inline provider/cluster/service.go]: should use bytes.Equal(ev.Deployment, dm.reservation.OrderID().DeploymentID()) instead

**Comment by boz** [comment]
adding more to this.

**Pull request closed without merging**